schemas/common/descriptors/relationshipdescriptor.schema.json

```diff
@@ -35,6 +35,19 @@
           "$ref": "https://ns.adobe.com/xdm/common/descriptors/itemselector#/definitions/selector"
         }
       },
+      "oneOf": [
```

Comment:
should this be anyOf or are we restricting it to oneOf?

Reply:
Hey @polavishnu4444, I'm not sure the registry tooling will handle this oneOf, @cmathis can you comment?

Reply:
@kstreeter 
This I assumed since we will be using it with some standard nodejs validator it would work.
Also today the all of merged result is having the oneOf values also at the top parallel to properties section.

Reply:
Hi @polavishnu4444 it's not so much the validation that's a problem, it's tooling like protobuf and POJO generation that breaks down.  We should probably just the properties optional at this point.  @cmathis please correct me if this won't be an issue

Reply:
@kstreeter if we make the properties optional, then there is not much of a value-add in this PR. The main intend of the change was to try and make sure the constraints between the various properties in a relationship descriptor are visible to clients and not hidden in the server-side code.

@cmathis do we filter-out 'oneof' usages when loading schemas from git? I do see some existing schemas that contain 'oneof' in this repo - https://github.com/adobe/xdm/search?p=1&q=oneof&unscoped_q=oneof

Reply:
Yeah,
I verified these in the local XDMR and also verified data payload with the schema generated (by combining the base descriptors fields) by XDMR with these definitions in global.

Reply:
Yeah Craig,
I verified these in the local XDMR and also verified data payload with the schema generated (by combining the base descriptors fields) by XDMR with these definitions in global.